vmdb/app/models/ems_amazon.rb

```diff
@@ -125,6 +125,20 @@ def vm_reboot_guest(vm, options = {})
     $log.error "MIQ(#{self.class.name}##{__method__}) vm=[#{vm.name}], error: #{err}"
   end
 
+  def stack_create(stack_name, template, options = {})
+    cloud_formation.stacks.create(stack_name, template.content, options).stack_id
+  rescue => err
+    $log.error "MIQ(#{self.class.name}##{__method__}) stack=[#{stack_name}], error: #{err}"
+    raise MiqException::MiqOrchestrationProvisionError, err.to_s
```

Comment:
The link says `cause` is a Ruby 2.1 feature which would not help us just yet.

Is the problem with provider specific errors the concern that the automate script will not know the error class?  If this is the case then I would recommend adding the stack trace to your raise command then:
`raise MiqException::MiqOrchestrationProvisionError, err.to_s, err.backtrace`